framework/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -371,16 +371,16 @@ def delete_node(self, owner_aid: str, node_id: int) -> None:
         with self.lock:
             if (
                 not (node := self.nodes.get(node_id))
-                or node.status == NodeStatus.DELETED
+                or node.status == NodeStatus.UNREGISTERED
                 or owner_aid != self.nodes[node_id].owner_aid
             ):
                 raise ValueError(
-                    f"Node ID {node_id} already deleted, not found or unauthorized "
-                    "deletion attempt."
+                    f"Node ID {node_id} already unregistered, not found or "
+                    "unauthorized deletion attempt."
```

Comment:
Wdyt? IMO, it's not "deletion" any more, and it reads a bit strange if we say "unauthorized unregistration attempt".

```suggestion
                    "the request was unauthorized."
```